### ISSUE TITLE
Combobox selectionBehavior=clear doesn't work

### ISSUE THREAD

**Issue by anilanar** [chakra-ui/zag#1277]
# 🐛 Bug report

## 💥 Steps to reproduce

2. Go to https://zagjs.com/components/react/combobox (0.36.3)
3. Set `selectionBehavior` to `clear`
4. Try selecting options.

It doesn't clear input after first interaction but clears it after clicking outside and clicking on combobox again.

## 🧐 Expected behavior

Combobox is cleared upon selection.

## 🌍 System information

| Software         | Version(s) |
| ---------------- | ---------- |
| Zag Version      |    5a199fdc8c9de339d766531bf70c584e652ccf3f       |
| Browser          |      chrome, firefox      |
| Operating System |     linux       |

## Reproduction
https://stackblitz.com/edit/vitejs-vite-gjlmnk?file=src%2Fcombobox.jsx

**Comment by TylerAPfledderer** [comment]
Hey @anilanar!

I'm unfortunately not seeing this negative behavior. I see you provided a screen recording in another PR. Could you do that for this one?

**Comment by spaaacetoast** [comment]
I've also just now noticed that this doesn't work, like its not immediately updating. only after a second interaction the input clears. Since I'm using SolidJs this issue might lie in the state machine

**Comment by anilanar** [comment]
@TylerAPfledderer Sure!

https://github.com/chakra-ui/zag/assets/753736/d72b3dd2-2161-4175-aca4-9447506cabdb
